server/application/rest/work_shift.py

```diff
@@ -65,18 +66,25 @@ def work_shifts():
         qrystr_params = {
             "filters": {},
         }
-        for arg, values in request.args.items():
-            print(arg, values)
+        facility_id = request.args.get("code")
+        if facility_id:
+            response_from_external_api = requests.get(f"https://api2-qa.gethelp.com/v2/facilities/{facility_id}")
+            if response_from_external_api.status_code == 200:
+                get_facility_info = response_from_external_api.json()
+                qrystr_params['get_facility_info'] = get_facility_info
+            else:
+                print("Failed to retrieve facility info from external API")
 
+        for arg, values in request.args.items():
             if arg.startswith("filter_"):
                 qrystr_params["filters"][arg.replace("filter_", "")] = values
-        print(qrystr_params)
-        # generate a request object
-        request_object = build_work_shift_list_request(
-            filters=qrystr_params["filters"]
-        )
-        # find workshifts matching the request object
+
+        request_object = build_work_shift_list_request(filters=qrystr_params["filters"])
         response = workshift_list_use_case(repo, request_object, user)
+
+        if 'get_facility_info' in qrystr_params:
+            response.value['get_facility_info'] = qrystr_params['get_facility_info']
```

Comment:
get_facility_info should not be in the parameters. Instead, you would loop through each work_shift that's in the response (the return value of the workshift_list_use_case), and get the associated facility id, and request info for this id (via the new use case that you should add). Append the facility info to the response, converted to JSON.